### ISSUE TITLE
Training does not converge for resnet50-ssd on pascal VOC dataset

### ISSUE THREAD

**Issue by kristellmarisse** [jay-mahadeokar/pynetbuilder#2]
I am traing SSD-Resnet50 on pascal VOC dataset. SInce I have a smaller GPU (gtx960 4gb), I reduced the batch size to train. The training loss started at 14 and after 7k iterations it went down to 7. But after that the loss doesn't seem to reduce. Is it because of changing the batch size ?

**Comment by jay-mahadeokar** [comment]
What is your batch size? I get best results with batch size of 32 (8 per gpu \* 4 gpus in parallel), also found that batch size as low as 14 also converges, though results are not best.  I would also look at running avg training loss to see whats happening, see the training [plot](https://github.com/jay-mahadeokar/pynetbuilder/tree/master/models/voc2007_ssd#residual-networks-based-detection-network) here.

**Comment by kristellmarisse** [comment]
Thank you for the leads. My batch size was only 2 (that was the best I can squeeze into my GPU memory). Is it ok if I increase the batch size by modifying the iter_size parameter in solver.prototxt. I usually use this trick in py-faster-rcnn.

**Comment by jay-mahadeokar** [comment]
@kristellmarisse I have not tried that setting. Maybe you could try using a smaller network for bigger batch size?  See few other resnet models shared [here](https://github.com/jay-mahadeokar/pynetbuilder/tree/master/models/imagenet) pretrained on imagenet, which give decent top-1 accuracy.  The # params field in the comparison tables will influence the model size.

**Comment by kristellmarisse** [comment]
Thank you for sharing more models.

**Comment by kristellmarisse** [comment]
By the way, can you share me your GPU specs on which you trained the Resnet+SSD?

**Comment by jay-mahadeokar** [comment]
I think its K80 it has 11GB memory.
